src/Std/Data/DHashMap/Internal/RawLemmas.lean

```diff
@@ -830,15 +833,525 @@ theorem contains_keys [EquivBEq α] [LawfulHashable α] (h : m.1.WF) {k : α} :
+    (h₁ : (l.map Sigma.fst).contains k = false)
+    {h'} :
+    (m.insertMany l).1.getKey k h' =
+    m.getKey k (contains_of_contains_insertMany_list _ h h' h₁) := by
+  simp_to_model using getKey_insertList_of_contains_eq_false
+
+theorem getKey_insertMany_list_of_mem [EquivBEq α] [LawfulHashable α] (h : m.1.WF)
+    {l : List ((a : α) × β a)}
+    {k k' : α} (k_beq : k == k')
+    (distinct : l.Pairwise (fun a b => (a.1 == b.1) = false))
+    (mem : k ∈ (l.map Sigma.fst))
+    {h'} :
+    (m.insertMany l).1.getKey k' h' = k := by
+  simp_to_model using getKey_insertList_of_mem
+
+theorem getKey!_insertMany_list_of_contains_eq_false [EquivBEq α] [LawfulHashable α] [Inhabited α]
+    (h : m.1.WF) {l : List ((a : α) × β a)} {k : α}
+    (h' : (l.map Sigma.fst).contains k = false) :
+    (m.insertMany l).1.getKey! k = m.getKey! k := by
+  simp_to_model using getKey!_insertList_of_contains_eq_false
+
+theorem getKey!_insertMany_listof_mem [EquivBEq α] [LawfulHashable α] [Inhabited α] (h : m.1.WF)
+    {l : List ((a : α) × β a)}
+    {k k' : α} (k_beq : k == k')
+    (distinct : l.Pairwise (fun a b => (a.1 == b.1) = false))
+    (mem : k ∈ (l.map Sigma.fst)) :
+    (m.insertMany l).1.getKey! k' = k := by
+  simp_to_model using getKey!_insertList_of_mem
+
+theorem getKeyD_insertMany_list_of_contains_eq_false [EquivBEq α] [LawfulHashable α] (h : m.1.WF)
+    {l : List ((a : α) × β a)} {k fallback : α}
+    (h' : (l.map Sigma.fst).contains k = false) :
+    (m.insertMany l).1.getKeyD k fallback = m.getKeyD k fallback := by
+  simp_to_model using getKeyD_insertList_of_contains_eq_false
+
+theorem getKeyD_insertMany_list_of_mem [EquivBEq α] [LawfulHashable α] (h : m.1.WF)
+    {l : List ((a : α) × β a)}
+    {k k' fallback : α} (k_beq : k == k')
+    (distinct : l.Pairwise (fun a b => (a.1 == b.1) = false))
+    (mem : k ∈ (l.map Sigma.fst)) :
+    (m.insertMany l).1.getKeyD k' fallback = k := by
+  simp_to_model using getKeyD_insertList_of_mem
+
+theorem size_insertMany_list [EquivBEq α] [LawfulHashable α] (h : m.1.WF)
+    {l : List ((a : α) × β a)} {distinct : List.Pairwise (fun a b => (a.1 == b.1) = false) l}:
+    (∀ (a : α), ¬ (m.contains a ∧ (l.map Sigma.fst).contains a)) →
+    (m.insertMany l).1.1.size = m.1.size + l.length := by
+  simp_to_model using length_insertList
+
+theorem isEmpty_insertMany_list_eq_false_of_isEmpty_eq_false [EquivBEq α] [LawfulHashable α]
+    (h : m.1.WF) {l : List ((a : α) × β a)}:
+    (m.1.isEmpty = false) → (m.insertMany l).1.1.isEmpty  = false := by
+  simp_to_model using isEmpty_insertList_eq_false_of_isEmpty_eq_false
+
+theorem isEmpty_insertMany_list [EquivBEq α] [LawfulHashable α] (h : m.1.WF)
+    {l : List ((a : α) × β a)} :
+    (m.insertMany l).1.1.isEmpty ↔ m.1.isEmpty ∧ l.isEmpty := by
+  simp_to_model using isEmpty_insertList
+
+section
+variable {β: Type v} (m: Raw₀ α (fun _ => β))
+
+theorem get?_insertMany_list_of_contains_eq_false_const [EquivBEq α] [LawfulHashable α] (h : m.1.WF)
```

Comment:
Okay, I see the problem with naming these lemmas. I think the (admittedly slighly crazy) scheme we use for the naming here would be that the lemma lives in the namespace of the first thing that appears in the name and then other relevant namespaces are included in the theorem name as appropriate. So this lemma would be called `Const.get?_insertMany_list_of_contains_eq_false` and the lemma you're currently calling `Const.get?_insertMany_list_of_contains_eq_false` would be called `Const.get?_constInsertMany_list_of_contains_eq_false`. Luckily, it's unlikely that end users will need to interact with these lemmas directly.